pkg/cmd/server/kubernetes/node_config.go

```diff
@@ -169,6 +171,32 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig) (*NodeConfig, error
 		return nil, err
 	}
 
+	// Replace the standard k8s emptyDir volume plugin with a wrapper version
+	// which offers XFS quota functionality, but only if the node config
+	// specifies an empty dir quota to apply to projects:
+	if options.ProjectEmptyDirQuota != nil {
+		glog.V(2).Info("Replacing empty-dir volume plugin with quota wrapper")
+		wrappedEmptyDirPlugin := false
+		for idx, plugin := range cfg.VolumePlugins {
+			// Can't really do type checking or use a constant here as they are not exported:
+			if plugin.Name() == "kubernetes.io/empty-dir" {
```

Comment:
> Does it really seem likely the plugin interface methods in use here are going to change during a minor or other rebase without us noticing?

Yes.  `RESTMapper` broke two weeks ago and it is used in every client command that we run.  It manifested as some broken shortcuts which were only caught by our e2e test.  I don't see corresponding tests here, so that would result in a slip and effectively disabling this feature silently.  And by and large, rebasers know more about `RESTMappers` than `VolumePlugins`.